test/Table_Tests/src/Formatting/Parse_Values_Spec.enso

```diff
@@ -395,22 +394,59 @@ spec =
             c1 = Column.from_vector "ints" ["0", "+0", "-0", "+1", "-1", "1", "000", "0010", "12345", Nothing]
             c2 = c1.parse type=Value_Type.Integer
             c2.name.should_equal c1.name
-            c2 . to_vector . should_equal [0, 0, 0, 1, -1, 1, Nothing, Nothing, 12345, Nothing]
+            c2 . to_vector . should_equal [0, 0, 0, 1, -1, 1, 0, 10, 12345, Nothing]
             c2.value_type.should_equal Value_Type.Integer
-            Problems.expect_warning Leading_Zeros c2
+            Problems.assume_no_problems c2
 
             c3 = c1.parse type=Value_Type.Integer format=(Data_Formatter.Value.with_number_formatting allow_leading_zeros=True)
             c3.to_vector . should_equal [0, 0, 0, 1, -1, 1, 0, 10, 12345, Nothing]
             Problems.assume_no_problems c3
 
+        Test.specify "should correctly parse integers in US formats" <|
+            cUS = Column.from_vector "ints" ["1", "000123", "-1234", "1234567", "123e456"]
+            pUS = cUS.parse type=Value_Type.Integer
+            pUS.to_vector . should_equal [1, 123, -1234, 1234567, Nothing]
+            Problems.expect_warning (Invalid_Format.Error "ints" Value_Type.Integer ["123e456"]) pUS
+
+            cUS2 = Column.from_vector "ints" ["1", "000123", "-1,234", "1,234,567", "12,34,56"]
+            pUS2 = cUS2.parse type=Value_Type.Integer
+            pUS2.to_vector . should_equal [1, 123, -1234, 1234567, Nothing]
+            Problems.expect_warning (Invalid_Format.Error "ints" Value_Type.Integer ["12,34,56"]) pUS2
+
+            cUS3 = Column.from_vector "ints" ["1", "(000,123)", "-1,234", "1,234,567", "12,34,56"]
+            pUS3 = cUS3.parse type=Value_Type.Integer
+            pUS3.to_vector . should_equal [1, -123, Nothing, 1234567, Nothing]
+            Problems.expect_warning (Invalid_Format.Error "ints" Value_Type.Integer ["-1,234", "12,34,56"]) pUS3
+
+            cUS4 = Column.from_vector "ints" ["$1234", "$1,234", "$1,234,567","-$1,234", "($1,234,567)"]
+            pUS4 = cUS4.parse type=Value_Type.Integer
+            pUS4.to_vector . should_equal [1234, 1234, 1234567, -1234, Nothing]
+            Problems.expect_warning (Invalid_Format.Error "ints" Value_Type.Integer ["($1,234,567)"]) pUS4
```

Comment:
These parentheses parsing rules look _really weird_. Maybe I just don't have enough knowledge about number formatting, but I assume other users may not know these cases either.

Maybe we can have a `? Number Formats` paragraph somewhere? Possibly in the `Data_Formatter`? So that if the user comes looking for 'why does this behave this weird way', they can find something.

Reply:
Basically the set of formats it can parse comes from going through the set Excel has.

The one I haven't done which we might want to add is allow `%` too.

Reply:
Ok, fair. I think it's stil worth documenting these supported formats somewhere - not everyone will know the more fancy Excel formats and since we are not affiliated with Excel it is not immediately clear we try to support the same formats.